### PR TITLE
Command line work start

### PR DESCRIPTION
**Pull request by mammerla**

### CODE REVIEW

File: app/src/app/ProjectExporter.ts [hunk 1]
```diff
@@ -410,6 +410,86 @@ export default class ProjectExporter {
...
   }
 
+  static async deployAsWorld(
+    carto: Carto,
+    project: Project,
+    worldProjectItem: ProjectItem,
+    returnZipBytes: boolean,
+    deployFolder?: IFolder
+  ) {
+    let mcworld: MCWorld | undefined = undefined;
```
[the remark this answers is not in the document]
Comment (mammerla) [reply app/src/app/ProjectExporter.ts:420]: Will fix that.  Just not used to seeing let mcworld: MCWorld | undefined; , it's like an unfinished sentence :)

File: app/src/cli/TaskWorker.ts [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+
+async function executeTask(task: ITask) {
+  if (!task.project) {
+    Log.error("Could not find an associated project for the associated task.");
+    return undefined;
+  }
+
+  activeContext = task.project?.ctorProjectName;
+
+  if (localEnv === undefined) {
```
[the remark this answers is not in the document]
Comment (mammerla) [reply app/src/cli/TaskWorker.ts:33]: After a lot of time spent using !localEnv and patterns like that, I find that bugs get introduced when localEnv could be 0 and this triggers the !foo check unintentionally 

(in this case it's always an object, so it doesn't really apply, but I'm trying to stay in the habit of being fully specific about "not 0" checks)

File: app/src/cli/ClUtils.ts [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+
+  static async getMainWorkFolder(taskType: TaskType, inputFolder?: string, outputFolder?: string) {
+    // console.log("Using local path: '" + inputFolder + "' from '" + __dirname + "'");
+    let workFolder: IFolder | undefined = undefined;
+
+    if (!inputFolder && outputFolder && ClUtils.getIsWriteCommand(taskType)) {
+      const outputStorage = new NodeStorage(outputFolder, "");
+      workFolder = outputStorage.rootFolder;
+      await workFolder.ensureExists();
+    }
```
[the remark this answers is not in the document]
Comment (mammerla) [reply app/src/cli/ClUtils.ts:101]: Good catch.  There might be writable commands in the future, though, so I'll keep this code for now

File: app/src/cli/index.ts
[the remark this answers is not in the document]
Comment (mammerla) [reply app/src/cli/index.ts:1]: Added

**Review by zachcampbellskyboxlabs — APPROVED** [review]

**Pull request merged**